app/javascript/App/components/NewsfeedPage/NewsList.js

```diff
@@ -1,31 +1,84 @@
 import React, { Component, Fragment } from 'react'
+import _ from 'lodash'
 import NewsListItemAnimated from './NewsListItemAnimated'
 import LoadingIndicator from '../LoadingIndicator'
 
 class NewsList extends Component {
   state = { initialRender: true }
+
+  constructor(props) {
+    super(props)
+    this.mountOnScrollHandler = this.mountOnScrollHandler.bind(this)
+    this.unmountOnScrollHandler = this.unmountOnScrollHandler.bind(this)
+  }
+
   componentDidMount() {
     setTimeout(() => {
       this.setState({ initialRender: false })
     }, 6000)
+
+    this.mountOnScrollHandler()
+  }
+
+  componentDidUpdate() {
+    if (!window.isMobile && !window.isTablet) {
+      const $newsFeed = $('#news-feed')
+      const timer = setInterval(() => {
+        if ($newsFeed.height() < $newsFeed.find('> div').get(0).clientHeight) {
+          clearInterval(timer)
+        } else {
+          this.props.scrollBottomToReducer(true)
+        }
+      }, 1000)
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmountOnScrollHandler()
+  }
+
+  mountOnScrollHandler() {
+    if (window.isMobile || window.isTablet) {
+      const throttled = _.throttle(this.props.onScrollNewsFeedMobile, 1000)
+      $(window).scroll(throttled)
+    } else {
+      const throttled = _.throttle(this.props.onScrollNewsFeedDesktop, 1000)
```

Comment:
I'd rather move the distance/buffer detection here rather than in the action creator (and not dispatch {type:null})